jetty-core/jetty-websocket/jetty-websocket-jetty-server/src/main/java/org/eclipse/jetty/websocket/server/ServerWebSocketContainer.java

```diff
@@ -86,6 +86,32 @@ public static ServerWebSocketContainer ensure(Server server, ContextHandler cont
         return container;
     }
 
+    /**
+     * <p>Returns the {@link ServerWebSocketContainer}, ensuring that
+     * it is available via {@link #get(Context)}.</p>
+     * <p>If the {@link ServerWebSocketContainer} is not already available,
+     * an instance is created, stored to be available via {@link #get(Context)}
+     * and returned.</p>
+     * <p>This method should be invoked during the setup of the
+     * {@link Handler} hierarchy.</p>
+     *
+     * @param server the {@link Server} object used to lookup common WebSocket components and store the {@link ServerWebSocketContainer}
+     * @return a non-{@code null} {@link ServerWebSocketContainer}
+     */
+    public static ServerWebSocketContainer ensure(Server server)
+    {
+        ServerWebSocketContainer container = get(server.getContext());
+        if (container == null)
+        {
+            WebSocketComponents components = WebSocketServerComponents.ensureWebSocketComponents(server);
+            WebSocketMappings mappings = new WebSocketMappings(components);
+            container = new ServerWebSocketContainer(mappings);
+            container.addBean(mappings);
+            server.getContext().setAttribute(WebSocketContainer.class.getName(), container);
+        }
+        return container;
```

Comment:
```suggestion
    return ensure(server, null);
```

then in the original method do:
```java
Context context = contextHandler == null ? server.getContext : contextHandler.getContext();
```